waku/common/utils/crypto.nim

```diff
@@ -0,0 +1,17 @@
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import libp2p/crypto/crypto as libp2pCrypto
+
+export libp2pCrypto
+
+type RngCtx = ref HmacDrbgContext
+
+var rng {.threadvar.}: RngCtx
```

Comment:
These "module variables" are often described as code smell. This is Nim's equivalent for the C/C++ global variables.

We should instantiate the RNG in the application top-level proc (in the `wakunode2` `main()` proc) and then pass it down as a constructor dependency argument. Similar to all the dependencies we create (e.g., Waku Archive drivers, Waku node, Peer manager, etc.).

-------

I recommend [this book](https://www.manning.com/books/dependency-injection-principles-practices-patterns) to understand the Dependency Injection patterns and good practices, and more importantly, the antipatterns that should be avoided. It is a .Net book, but the important part is the principles.